test/fgpu/test_engine.py

```diff
@@ -608,9 +633,12 @@ async def test_delay_phase_rate(
         # One tone at centre frequency to test the absolute phase, and one at another
         # frequency to test the slope across the band.
         tone_channels = [CHANNELS // 2, CHANNELS - 123]
-        tones = [CW(frac_channel=frac_channel(output, channel), magnitude=110) for channel in tone_channels]
+        tones = [
+            CW(frac_channel=frac_channel(output, channel), magnitude=110, delay=extra_delay_samples, phase=extra_phase)
+            for channel in tone_channels
+        ]
         src_layout = engine_server.src_layout
-        n_samples = 10 * src_layout.chunk_samples
+        n_samples = 32 * src_layout.chunk_samples
```

Comment:
I didn't approach it too scientifically. I was getting an error because with bigger output chunks and bigger window sizes, there wasn't enough data for the test to work, and so I increased it until I stopped getting the error.